packages/video_player/closed_caption_file/lib/closed_caption_file.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library closed_caption_file;
+
+import 'src/sub_rip.dart';
+
+/// A structured representation of a parsed closed caption file.
+///
+/// A closed caption file includes a list of captions, each with a start and end
+/// time for when the given closed caption should be displayed.
+///
+/// The [captions] are a list of all captions in a file, in the order that they
+/// appeared in the file.
+class ClosedCaptionFile {
+  /// The full list of captions from a given file.
+  final List<Caption> captions;
+
+  /// Creates a dart representation of an entire file's worth of closed
+  /// captioning.
+  ///
+  /// This constructor should only be used if you're writing a parser for a
+  /// currently unsupported file type.
+  ///
+  /// Prefer using one of the convenience constructors, such as
+  /// `ClosedCaptionFile.fromSubRipFile(...)`.
+  const ClosedCaptionFile({this.captions = const <Caption>[]});
```

Comment:
subjective, optional: I think it would be a little more idiomatic to write `ClosedCaptionFile` as an abstract class or a mixin, and create another class like `SubRipCaption` that uses it.

Reply:
Good call, I like that better, thanks for the suggestion.

Reply:
Good call, I like that better, thanks for the suggestion.